table/block_based_table_reader.cc

```diff
@@ -208,10 +178,15 @@ class PartitionIndexReader : public IndexReader {
   // return a two-level iterator: first level is on the partition index
   virtual InternalIterator* NewIterator(BlockIter* iter = nullptr,
                                         bool dont_care = true) override {
+    // Filters are already checked before seeking the index
+    const bool skip_filters = true;
+    const bool is_user_data = true;
     return NewTwoLevelIterator(
-        new BlockBasedTable::BlockEntryIteratorState(table_, ReadOptions(),
-                                                     false),
-        index_block_->NewIterator(comparator_, iter, true));
+        new BlockBasedTable::BlockEntryIteratorState(
+            table_, ReadOptions(), skip_filters, !is_user_data),
+        index_block_->NewIterator(comparator_, nullptr, true));
+    // TODO: Update TwoLevelIterator to be able to make use of on-stack
+    // BlockIter while the state is on heap
```

Comment:
Sure, we can leave it for future since it is not introduce in this patch. It is said that "premature optimization is the root of all evil" and I want to avoid that.